<launch>

  <!-- General Parameters -->
  <arg name="world_frame_name" value="world"/>
  
  <!-- Goal Parameters -->
  <arg name="goal_msg" value="/goal"/>

  <!-- Robot Parameters -->

  <!--training_garden_static_0
      training_mav_garden_static_0
      training_garden_dynamic_0
      cylinders0-->
  <arg name="world_name" value=""/>
  
  <!--turtlebot0-->
  <arg name="robot_name" value="turtlebot0"/>
  <arg name="robot_frame_name" value="turtlebot0/base_link"/>
  <arg name="robot_bbx_x_max" value="0.2"/>                   <!--NUA: Unnecessary? Parse URDF? -->
  <arg name="robot_bbx_x_min" value="-0.2"/>                   <!--NUA: Unnecessary? Parse URDF? -->
  <arg name="robot_bbx_y_max" value="0.2"/>                   <!--NUA: Unnecessary? Parse URDF? -->
  <arg name="robot_bbx_y_min" value="-0.2"/>                   <!--NUA: Unnecessary? Parse URDF? -->
  <arg name="robot_bbx_z_max" value="0.4"/>                  <!--NUA: Unnecessary? Parse URDF? -->
  <arg name="robot_bbx_z_min" value="-0.0"/>                  <!--NUA: Unnecessary? Parse URDF? -->
  <arg name="robot_min_lat_velo" value="0.0"/>         <!--NUA: Unnecessary? -->
  <arg name="robot_max_lat_velo" value="0.4"/>
  <arg name="robot_max_lat_acc" value="2.0"/>
  <arg name="robot_min_yaw_velo" value="-2.0"/>        <!--NUA: Unnecessary!  -->
  <arg name="robot_max_yaw_velo" value="2.0"/>
  <arg name="robot_max_yaw_acc" value="0.0"/>
  <arg name="robot_init_pos_x" value="0.0"/>         <!--NUA: Unnecessary? Use TF? -->
  <arg name="robot_init_pos_y" value="0.0"/>           <!--NUA: Unnecessary? Use TF? -->
  <arg name="robot_init_pos_z" value="0.0"/>            <!--NUA: Unnecessary? Use TF? -->
  <arg name="robot_init_yaw" value="$(eval 0)"/>     <!--NUA: Unnecessary? Use TF? -->
  <arg name="robot_pose_control_msg" value=""/>
  <arg name="robot_velo_control_msg" value=""/>
  <arg name="odometry_msg" value="/turtlebot0/odom"/>
  <arg name="map_msg" value="/server_octomap"/>

  <!-- turtlebot0 Process Parameters -->
  <arg name="visu_flag" value="true" />
  <arg name="time_limit" value="0" />
  <arg name="dt" value="0.1" />
  <arg name="goal_close_threshold" value="0.25" />

  <!-- turtlebot0 Trajectory Sampling Parameters-->
  <arg name="trajectory_sampling_dataset_path" value="/home/akmandor/catkin_ws/src/motion_planning/tentabot/dataset/trajectory_sampling/" />
  <arg name="trajectory_gen_type" value="kinematic" />
  <arg name="ttime" value="10.0" />
  <arg name="tlen" value="10" />
  <arg name="tyaw" value="0.0" />
  <arg name="tpitch" value="0.0" />
  <arg name="tsamp_cnt" value="20" />
  <arg name="lat_velo_samp_cnt" value="11" />
  <arg name="tyaw_cnt" value="13" />
  <arg name="tpitch_cnt" value="1" />

  <!-- Offline Tuning Parameters -->
  <!--arg name="trajectory_data_path" value="" /-->
  <arg name="trajectory_data_path" value="/home/akmandor/catkin_ws/src/motion_planning/tentabot/dataset/trajectory_sampling/20211011_150548/" />

  <arg name="max_occupancy_belief_value" value="100" />
  <arg name="pdist_x_max" value="0.2" />
  <arg name="pdist_x_min" value="-0.2" />
  <arg name="pdist_y_max" value="0.2" />
  <arg name="pdist_y_min" value="-0.2" />
  <arg name="pdist_z_max" value="0.3" />
  <arg name="pdist_z_min" value="-0.0" />
  <arg name="sdist_x_max" value="0.4" />
  <arg name="sdist_x_min" value="-0.4" />
  <arg name="sdist_y_max" value="0.4" />
  <arg name="sdist_y_min" value="-0.4" />
  <arg name="sdist_z_max" value="0.5" />
  <arg name="sdist_z_min" value="-0.0" />
  <arg name="sweight_max" value="1" />
  <arg name="egrid_vdim" value="0.1" />

  <!-- Online Tuning Parameters -->
  <arg name="tbin_obs_cnt_threshold" value="1" />
  <arg name="crash_dist_scale" value="0.5" />

  <!-- Data Collection Parameters -->
  <arg name="nav_data_path" value="" />
  <!--arg name="nav_data_path" value="/home/akmandor/catkin_ws/src/motion_planning/tentabot/dataset/tnav/" /-->

  <!-- Simulation Parameters -->
  <arg name="ns" default="/" />

  <group ns="$(arg ns)">

    <!-- LET'S GO -->
    <node name="tentabot_drl_service" pkg="tentabot" type="tentabot_drl_service" output="screen">   
      <param name="world_frame_name" value="$(arg world_frame_name)" />
      <param name="goal_msg" value="$(arg goal_msg)" />
      <param name="world_name" value="$(arg world_name)" />
      <param name="robot_name" value="$(arg robot_name)" />
      <param name="robot_frame_name" value="$(arg robot_frame_name)" />
      <param name="robot_bbx_x_max" value="$(arg robot_bbx_x_max)" />
      <param name="robot_bbx_x_min" value="$(arg robot_bbx_x_min)" />
      <param name="robot_bbx_y_max" value="$(arg robot_bbx_y_max)" />
      <param name="robot_bbx_y_min" value="$(arg robot_bbx_y_min)" />
      <param name="robot_bbx_z_max" value="$(arg robot_bbx_z_max)" />
      <param name="robot_bbx_z_min" value="$(arg robot_bbx_z_min)" />
      <param name="robot_min_lat_velo" value="$(arg robot_min_lat_velo)" />
      <param name="robot_max_lat_velo" value="$(arg robot_max_lat_velo)" />
      <param name="robot_max_lat_acc" value="$(arg robot_max_lat_acc)" />
      <param name="robot_min_yaw_velo" value="$(arg robot_min_yaw_velo)" />
      <param name="robot_max_yaw_velo" value="$(arg robot_max_yaw_velo)" />
      <param name="robot_max_yaw_acc" value="$(arg robot_max_yaw_acc)" />
      <param name="robot_init_pos_x" value="$(arg robot_init_pos_x)" />
      <param name="robot_init_pos_y" value="$(arg robot_init_pos_y)" />
      <param name="robot_init_pos_z" value="$(arg robot_init_pos_z)" />
      <param name="robot_init_yaw" value="$(arg robot_init_yaw)" />
      <param name="robot_pose_control_msg" value="$(arg robot_pose_control_msg)" />
      <param name="robot_velo_control_msg" value="$(arg robot_velo_control_msg)" />
      <param name="odometry_msg" value="$(arg odometry_msg)" />
      <param name="map_msg" value="$(arg map_msg)" />
      <!--param name="occupancy_sensor_msg" value="$(arg occupancy_sensor_msg)" /-->
      
      <param name="visu_flag" value="$(arg visu_flag)" />
      <param name="time_limit" value="$(arg time_limit)" />
      <param name="dt" value="$(arg dt)" />
      <param name="goal_close_threshold" value="$(arg goal_close_threshold)" />
      
      <param name="trajectory_sampling_dataset_path" value="$(arg trajectory_sampling_dataset_path)" />
      <param name="trajectory_gen_type" value="$(arg trajectory_gen_type)" />
      <param name="ttime" value="$(arg ttime)" />
      <param name="tlen" value="$(arg tlen)" />
      <param name="tyaw" value="$(arg tyaw)" />
      <param name="tpitch" value="$(arg tpitch)" />
      <param name="tsamp_cnt" value="$(arg tsamp_cnt)" />
      <param name="lat_velo_samp_cnt" value="$(arg lat_velo_samp_cnt)" />
      <param name="tyaw_cnt" value="$(arg tyaw_cnt)" />
      <param name="tpitch_cnt" value="$(arg tpitch_cnt)" />
      
      <param name="trajectory_data_path" value="$(arg trajectory_data_path)" />
      <param name="max_occupancy_belief_value" value="$(arg max_occupancy_belief_value)" />
      <param name="pdist_x_max" value="$(arg pdist_x_max)" />
      <param name="pdist_x_min" value="$(arg pdist_x_min)" />
      <param name="pdist_y_max" value="$(arg pdist_y_max)" />
      <param name="pdist_y_min" value="$(arg pdist_y_min)" />
      <param name="pdist_z_max" value="$(arg pdist_z_max)" />
      <param name="pdist_z_min" value="$(arg pdist_z_min)" />
      <param name="sdist_x_max" value="$(arg sdist_x_max)" />
      <param name="sdist_x_min" value="$(arg sdist_x_min)" />
      <param name="sdist_y_max" value="$(arg sdist_y_max)" />
      <param name="sdist_y_min" value="$(arg sdist_y_min)" />
      <param name="sdist_z_max" value="$(arg sdist_z_max)" />
      <param name="sdist_z_min" value="$(arg sdist_z_min)" />
      <param name="sweight_max" value="$(arg sweight_max)" />
      <param name="egrid_vdim" value="$(arg egrid_vdim)" />
      
      <param name="tbin_obs_cnt_threshold" value="$(arg tbin_obs_cnt_threshold)" />
      <param name="crash_dist_scale" value="$(arg crash_dist_scale)" />

      <param name="nav_data_path" value="$(arg nav_data_path)" />
    </node>
  
  </group>
</launch>